geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/MessageDispatcher.java

```diff
@@ -581,6 +656,33 @@ protected boolean dispatchMessage(ClientMessage clientMessage) throws IOExceptio
         logger.trace(msg.toString());
       }
 
+      // authorize the message before dispatching
+      String regionName = clientUpdateMessage.getRegionName();
+      Object key = clientUpdateMessage.getKeyOfInterest();
+      SecurityService securityService = getCache().getSecurityService();
+      ResourcePermission permission = new ResourcePermission(ResourcePermission.Resource.DATA,
+          ResourcePermission.Operation.READ,
+          regionName, key == null ? null : key.toString());
```

Comment:
`permission` isn't needed outside the body of the next `if`. And even then it is only needed conditionally (sometimes it isn't needed at all). And `regionName` and `key` are used only to compute `permission`.

Please consider replacing lines 660, 661, 663-665 w/ a method that takes `clientUpdateMessage` and returns `ResourcePermission` and then call that method in the two places `permission` is currently used below. This will eliminate some spurious object allocation and make this code a little simpler.

Reply:
done